crates/miden-lib/src/account/wallets/mod.rs

```diff
@@ -59,8 +62,9 @@ pub fn create_basic_wallet(
         ));
     }
 
-    let auth_component: RpoFalcon512 = match auth_scheme {
-        AuthScheme::RpoFalcon512 { pub_key } => RpoFalcon512::new(pub_key),
+    let auth_component: AccountComponent = match auth_scheme {
+        AuthScheme::RpoFalcon512 { pub_key } => RpoFalcon512::new(pub_key).into(),
+        AuthScheme::NoAuth => NoAuth::new().into(),
```

Comment:
Not for this PR, but I wonder if we should allow this. I at some point in the future it could make sense for low-value "on-device" private wallets. But also, could be a potential footgun. Let's create an issue (or discussion) to discuss this.